pkgs/tools/package-management/poetry/default.nix

```
@@ -1,39 +1,75 @@
 { lib
 , python3
 , fetchFromGitHub
+, fetchPypi
 }:
 
 let
   python = python3.override {
     packageOverrides = self: super: {
       poetry = self.callPackage ./unwrapped.nix { };
 
+      filelock = super.filelock.overridePythonAttrs (old: rec {
+        version = "3.12.2";
+        src = fetchPypi {
+          inherit (old) pname;
+          inherit version;
+          hash = "sha256-ACdAUY2KpZomsMduEPuMbhXq6CXTS2/fZwMz/XuTjYE=";
+        };
+        nativeCheckInputs = with self; [
+          pytest-mock
+          pytestCheckHook
+        ];
+      });
       # version overrides required by poetry and its plugins
       platformdirs = super.platformdirs.overridePythonAttrs (old: rec {
-        version = "2.6.2";
+        version = "3.8.0";
         src = fetchFromGitHub {
           owner = "platformdirs";
           repo = "platformdirs";
           rev = "refs/tags/${version}";
-          hash = "sha256-yGpDAwn8Kt6vF2K2zbAs8+fowhYQmvsm/87WJofuhME=";
+          hash = "sha256-eQAEHl61aC/B44G3zBqMjDVAQF8GerpJbeQ1nT4uQ4Q=";
         };
         SETUPTOOLS_SCM_PRETEND_VERSION = version;
       });
       poetry-core = super.poetry-core.overridePythonAttrs (old: rec {
-        version = "1.5.2";
+        version = "1.6.1";
         src = fetchFromGitHub {
           owner = "python-poetry";
           repo = "poetry-core";
           rev = version;
-          hash = "sha256-GpZ0vMByHTu5kl7KrrFFK2aZMmkNO7xOEc8NI2H9k34=";
+          hash = "sha256-Gc22Y2T4uO39jiOqEUFeOfnVCbknuDjmzFPZgk2eY74=";
+        };
+        nativeCheckInputs = old.nativeCheckInputs ++ [
+          self.tomli
+          self.tomli-w
+        ];
```

```suggestion
        nativeCheckInputs = old.nativeCheckInputs ++ [
          self.tomli-w
        ];
```

It doesn't look like `poetry-core` depends on `tomli`, it only depends on `tomli-w`:
https://github.com/python-poetry/poetry-core/blob/0e16061b49decb4adf49df09309bc6d8f2c94e75/pyproject.toml#L42